lib/dry/system/component_dir.rb

```diff
@@ -50,6 +50,15 @@ def component_file(component_path)
         component_file if component_file.exist?
       end
 
+      # @api private
+      def component_options
+        {}.tap do |opts|
+          opts[:auto_register] = config.auto_register
+          opts[:loader] = config.loader if config.loader
```

Comment:
This conditional bothers me. One way to avoid it would be to propagate the top-level configured `loader` to all component_dirs that don't have a custom one configured. Or alternatively, to remove the top-level `loader` setting and have it live underneath component dirs only.